Panic in `pagerduty_enablement` resource: nil pointer dereference during Create operation

**Terraform Version:** (Please specify)
**Provider Version:** v3.30.0
**Affected Resources:** 
- `pagerduty_enablement`
- `pagerduty_event_orchestration_router`

### Expected Behavior
The `pagerduty_enablement` resource should create successfully without causing a plugin crash.

### Actual Behavior
The provider crashes with a nil pointer dereference panic during the Create operation for the `pagerduty_enablement` resource.

### Steps to Reproduce
1. Configure a `pagerduty_enablement` resource block for event orchestrations
2. Run `terraform apply`
3. Provider crashes with segmentation violation

### Configuration
The issue occurs when using resource blocks for enabling AIOps on services and orchestrations. Initially encountered with both services and orchestrations enabled, but persists when only event orchestrations enablement is configured.

### Error Output
```
Error: Plugin did not respond

  with module.pd_stack.module.pd_team["qrt-tech-ops"].pagerduty_event_orchestration_router.router["Grafana"],
  on ../../modules/main.tf line 68, in resource "pagerduty_event_orchestration_router" "router":
  68: resource "pagerduty_event_orchestration_router" "router" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
contain more details.

Error: Plugin did not respond

  with module.pd_stack.module.pd_team["qrt-tech-ops"].pagerduty_enablement.catchall,
  on ../../modules/main.tf line 122, in resource "pagerduty_enablement" "catchall":
 122: resource "pagerduty_enablement" "catchall" {

The plugin encountered an error, and failed to respond to the
plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
contain more details.
```

### Stack Trace
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0xf3f593]

goroutine 137 [running]:
github.com/PagerDuty/terraform-provider-pagerduty/pagerdutyplugin.(*resourceEnablement).Create.func1()
  github.com/PagerDuty/terraform-provider-pagerduty/pagerdutyplugin/resource_pagerduty_enablement.go:118 +0x253
github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.RetryContext.func1()
  github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/wait.go:30 +0x4e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext.func1()
  github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/state.go:113 +0x1a3
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry.(*StateChangeConf).WaitForStateContext in goroutine 135
  github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/retry/state.go:86 +0x237
```

### Analysis
The panic occurs at line 118 in `resource_pagerduty_enablement.go` within the Create function, suggesting a nil pointer is being dereferenced during the enablement creation process. This appears to be related to AIOps enablement functionality.

### Additional Context
- The issue persists whether AIOps is enabled for both services and orchestrations or just orchestrations
- Multiple resources are affected by the same underlying issue
- This is a critical bug as it prevents successful Terraform operations

--- a/pagerdutyplugin/resource_pagerduty_enablement.go
+++ b/pagerdutyplugin/resource_pagerduty_enablement.go
@@ -80,11 +80,12 @@ func (r *resourceEnablement) Create(ctx context.Context, req resource.CreateRequ
 	// API may return warnings if account is not entitled to AIOps features but the setting is still updated
 	err := retry.RetryContext(ctx, 2*time.Minute, func() *retry.RetryError {
 		var enablementResult *pagerduty.Enablement
+		var result *pagerduty.EnablementWithWarnings
 		var err error
 
 		switch enablement.EntityType {
 		case "service":
-			result, err := r.client.UpdateServiceEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, enablement.Enabled)
+			result, err = r.client.UpdateServiceEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, enablement.Enabled)
 			if err == nil {
 				for _, warning := range result.Warnings {
 					resp.Diagnostics.AddWarning("PagerDuty API Warning", warning)
@@ -94,7 +95,7 @@ func (r *resourceEnablement) Create(ctx context.Context, req resource.CreateRequ
 				enablementResult = nil
 			}
 		case "event_orchestration":
-			result, err := r.client.UpdateEventOrchestrationEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, enablement.Enabled)
+			result, err = r.client.UpdateEventOrchestrationEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, enablement.Enabled)
 			if err == nil {
 				for _, warning := range result.Warnings {
 					resp.Diagnostics.AddWarning("PagerDuty API Warning", warning)
@@ -279,11 +280,12 @@ func (r *resourceEnablement) Update(ctx context.Context, req resource.UpdateRequ
 	// API may return warnings if account is not entitled to AIOps features but the setting is still updated
 	err := retry.RetryContext(ctx, 2*time.Minute, func() *retry.RetryError {
 		var enablementResult *pagerduty.Enablement
+		var result *pagerduty.EnablementWithWarnings
 		var err error
 
 		switch enablement.EntityType {
 		case "service":
-			result, err := r.client.UpdateServiceEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, enablement.Enabled)
+			result, err = r.client.UpdateServiceEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, enablement.Enabled)
 			if err == nil {
 				for _, warning := range result.Warnings {
 					resp.Diagnostics.AddWarning("PagerDuty API Warning", warning)
@@ -293,7 +295,7 @@ func (r *resourceEnablement) Update(ctx context.Context, req resource.UpdateRequ
 				enablementResult = nil
 			}
 		case "event_orchestration":
-			result, err := r.client.UpdateEventOrchestrationEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, enablement.Enabled)
+			result, err = r.client.UpdateEventOrchestrationEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, enablement.Enabled)
 			if err == nil {
 				for _, warning := range result.Warnings {
 					resp.Diagnostics.AddWarning("PagerDuty API Warning", warning)
@@ -348,18 +350,19 @@ func (r *resourceEnablement) Delete(ctx context.Context, req resource.DeleteRequ
 
 	// Disable the enablement by setting enabled=false
 	err := retry.RetryContext(ctx, 2*time.Minute, func() *retry.RetryError {
+		var result *pagerduty.EnablementWithWarnings
 		var err error
 
 		switch enablement.EntityType {
 		case "service":
-			result, err := r.client.UpdateServiceEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, false)
+			result, err = r.client.UpdateServiceEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, false)
 			if err == nil {
 				for _, warning := range result.Warnings {
 					resp.Diagnostics.AddWarning("PagerDuty API Warning", warning)
 				}
 			}
 		case "event_orchestration":
-			result, err := r.client.UpdateEventOrchestrationEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, false)
+			result, err = r.client.UpdateEventOrchestrationEnablementWithContext(ctx, enablement.EntityID, enablement.Feature, false)
 			if err == nil {
 				for _, warning := range result.Warnings {
 					resp.Diagnostics.AddWarning("PagerDuty API Warning", warning)

--- a/pagerdutyplugin/resource_pagerduty_enablement_test.go
+++ b/pagerdutyplugin/resource_pagerduty_enablement_test.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 	"log"
+	"regexp"
 	"strings"
 	"testing"
 
@@ -458,4 +459,60 @@ resource "pagerduty_enablement" "test_aiops" {
   enabled     = false
 }
 `, username, email, escalationPolicy, serviceName)
+}
+
+// TestAccPagerDutyEnablement_InvalidServiceError tests error handling
+// when attempting to create an enablement with an invalid service ID.
+// This test validates the fix for the variable shadowing bug that caused
+// nil pointer panics when API calls returned errors.
+func TestAccPagerDutyEnablement_InvalidServiceError(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: testAccProtoV5ProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckPagerDutyEnablementInvalidServiceConfig(),
+				ExpectError: regexp.MustCompile("(?i)(not found|invalid|does not exist|404|failed to find)"),
+			},
+		},
+	})
+}
+
+// TestAccPagerDutyEnablement_InvalidEventOrchestrationError tests error handling
+// when attempting to create an enablement with an invalid event orchestration ID.
+// This test validates the fix for the variable shadowing bug that caused
+// nil pointer panics when API calls returned errors.
+func TestAccPagerDutyEnablement_InvalidEventOrchestrationError(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: testAccProtoV5ProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccCheckPagerDutyEnablementInvalidEventOrchestrationConfig(),
+				ExpectError: regexp.MustCompile("(?i)(not found|invalid|does not exist|404|failed to find)"),
+			},
+		},
+	})
+}
+
+func testAccCheckPagerDutyEnablementInvalidServiceConfig() string {
+	return `
+resource "pagerduty_enablement" "test" {
+  entity_type = "service"
+  entity_id   = "PINVALID"
+  feature     = "aiops"
+  enabled     = true
+}
+`
+}
+
+func testAccCheckPagerDutyEnablementInvalidEventOrchestrationConfig() string {
+	return `
+resource "pagerduty_enablement" "test" {
+  entity_type = "event_orchestration"
+  entity_id   = "PINVALID"
+  feature     = "aiops"
+  enabled     = true
+}
+`
 }

